src/System.Linq.Expressions/src/System/Runtime/CompilerServices/ReadOnlyCollectionBuilder.cs

```diff
@@ -388,12 +396,20 @@ void IList.Remove(object value)
 
         void ICollection.CopyTo(Array array, int index)
         {
-            if (array == null)
-                throw new ArgumentNullException(nameof(array));
-            if (array.Rank != 1)
-                throw new ArgumentException(nameof(array));
+            if ((array != null) && (array.Rank != 1))
+            {
+                throw new ArgumentException(); // Arg_RankMultiDimNotSupported
```

Comment:
SR does not contain such or similar message. Also I tried add string to Strings.rex file but Visual Studio does not open it and return some moniker error. 
I will create issue about this after merge and somebody who knows how to add resource strings will make such changes later.